### DISCUSSION TITLE
vim 需要那些常规配置才能做到比VSCode 更加高效？

### DISCUSSION THREAD

**Discussion by Linya-IronMan** [nshen/learn-neovim-lua#44] [Q&A] [answered]
在插入模式下，想要回到行首就比较麻烦。
1. esc 会退到 normal 模式
2. 0 跳到行首
3. a/i 进入插入模式并开始输入

VSCode：
1. cmd +  <- /   home 键 就可以回到行首直接开始输入

除了这个例子，还有很多其他的情况都是一样。想要在输入模式下移动光标就相当地麻烦，这种情况应该如何配置快捷键？

要求能够在输入模式下就能完成所需要的光标移动
1. 单词跳转
2. 行选择
3. 单词选择
4. 行顺序移动
5. 光标行首/行尾的移动

从插入模式退出到normal模式下，完成移动后再进入插入模式，就算熟悉之后操作变快了，也是比VSCode多了两步，所以vim的高效率是从哪里来的呢？

**Comment by nshen** [comment, marked as answer]
我觉得还是熟悉的问题，你可能是用vscode 太久了，已经肌肉记忆了，我在vim下比用vscode快多了。

vim 推荐应该所有移动光标相关的都在 `normal` 模式下。我看有些人设置 `jk` 或 `jj` 快速回 normal 模式，而我是设置了 Cap 和 ESC 交换，现在点击 Cap 键已经植入我的肌肉记忆了，在同事的电脑上很难控制自己不敲Cap，根本没法敲代码。

在行首和行尾输入，应该在`normal`模式下直接`shift + i` 和 `shift + a` ，这两个我非常常用。
单词跳转我用 `w` 和 `b` 最多。
行选择 `shift + v`
单词选择 `viw`
跳到行首行尾我是 `^` 和 `$`  和正则表达式一致， 不是很常用。

**Comment by kang8** [comment]
> 在插入模式下，想要回到行首就比较麻烦。

我是直接设置了 emacs 键位，使用 control + a 在插入模式回到行首。非常方便。

```vim
inoremap <c-a> <home>
```
